From 54376b2431f48f81f1eba90564351770d315def6 Mon Sep 17 00:00:00 2001
From: David Reaver <me@davidreaver.com>
Date: Tue, 4 Feb 2025 18:59:06 -0800
Subject: [RFC PATCH 2/7] debugfs: Add #includes so debugfs_node is available
 in some files
To: Greg Kroah-Hartman <gregkh@linuxfoundation.org>,
    Rafael J. Wysocki <rafael@kernel.org>,
    Danilo Krummrich <dakr@kernel.org>
Cc: Steven Rostedt <rostedt@goodmis.org>,
    Christian Brauner <brauner@kernel.org>,
    Alexander Viro <viro@zeniv.linux.org.uk>,
    Tejun Heo <tj@kernel.org>,
    linux-fsdevel@vger.kernel.org,
    cocci@inria.fr,
    linux-kernel@vger.kernel.org,
    David Reaver <me@davidreaver.com>

The next commit in this series runs Coccinelle to automatically replace
dentry with debugfs_node. In order for that commit to compile, we need to
strategically place some #include <linux/debugfs.h> lines in some files.

Simply adding "struct debugfs_node;" is sometimes not enough. If both
dentry and debugfs_node are forward declared, but we don't have the
"#define debugfs_node dentry" included, they will be treated as distinct
types, defeating the purpose of the #define.

Signed-off-by: David Reaver <me@davidreaver.com>
---
 include/linux/fault-inject.h | 1 +
 include/linux/irqdesc.h      | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/linux/fault-inject.h b/include/linux/fault-inject.h
index 8c829d28dcf3..ab7ddd6fdc47 100644
--- a/include/linux/fault-inject.h
+++ b/include/linux/fault-inject.h
@@ -4,6 +4,7 @@
 
 #include <linux/err.h>
 #include <linux/types.h>
+#include <linux/debugfs.h>
 
 struct dentry;
 struct kmem_cache;
diff --git a/include/linux/irqdesc.h b/include/linux/irqdesc.h
index fd091c35d572..549d930407fd 100644
--- a/include/linux/irqdesc.h
+++ b/include/linux/irqdesc.h
@@ -5,6 +5,7 @@
 #include <linux/rcupdate.h>
 #include <linux/kobject.h>
 #include <linux/mutex.h>
+#include <linux/debugfs.h>
 
 /*
  * Core internal functions to deal with irq descriptors
